src/promptflow/promptflow/_core/tools_manager.py

```diff
@@ -249,6 +251,54 @@ def _load_custom_tool(tool: Tool, node_name: str) -> Callable:
         return f_globals[func_name]
 
 
+class ToolLoader:
+    def __init__(self, working_dir: str, package_tool_keys: Optional[List[str]] = None) -> None:
+        self._working_dir = working_dir
+        self._package_tools = collect_package_tools(package_tool_keys) if package_tool_keys else {}
+
+    def load_tool_for_node(self, node: Node) -> Tool:
+        if node.source is None:
+            raise UserErrorException(f"Node {node.name} does not have source defined.")
+        if node.type is ToolType.PYTHON:
```

Comment:
We should use `==` to replace `is` here.

See https://stackoverflow.com/a/133024 for details.

Reply:
Sorry, I was wrong. Enums can be compared using `is`.

https://stackoverflow.com/a/42866806

Reply:
Fixed.